content/en/docs/concepts/configuration/configmap.md

```diff
@@ -0,0 +1,78 @@
+---
+title: ConfigMaps
+content_template: templates/concept
+weight: 20
+---
+
+{{% capture overview %}}
+
+{{< glossary_definition term_id="configmap" prepend="A ConfigMap is" length="all" >}}
+
+{{< caution >}}
+ConfigMap does not provide secrecy or encryption.
+
+If the data you want to store are confidential, use a
+{{< glossary_tooltip text="Secret" term_id="secret" >}} rather than a ConfigMap,
+or use additional (third party) tools to keep your data private.
+{{< /caution >}}
+
+{{% /capture %}}
+
+{{% capture body %}}
+## ConfigMap object
+
+A ConfigMap is an API [object](/docs/concepts/overview/working-with-objects/kubernetes-objects/)
+that lets you store configuration for other objects to use. Unlike most
+Kubernetes objects that have a `spec`, a ConfigMap has a `data` section where you
+can place items (keys) and their values.
+
+## ConfigMaps and Pods
+There are three different ways that you can use a ConfigMap to configure
+a container inside a Pod:
+
+- Command line arguments to the entrypoint of a container
+- Environment variables for a container
+- Add a file in read-only volume, for the application to read
+
+These different methods lend themselves to different ways of modeling
+the data being consumed.
+
+Here's an example ConfigMap that has some keys with single values,
+and other keys where the value looks like a fragment of a configuration
+format.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  Name: example-configmap
+data:
+  # property-like keys; each key maps to a simple value
+  game-properties-file-name: game.properties
+  ui-properties-file-name: ui.properties
+  #
+  # file-like keys
+  game.properties: |
+    enemies=aliens
+    lives=3
+    enemies.cheat=true
+    enemies.cheat.level=noGoodRotten
+  ui.properties: |
+    color.good=purple
+    color.bad=yellow
+    allow.textmode=true
+```
+
+The ConfigMap itself doesn't treat those keys and values any differently
+if they look like property values or like configuration file data. The
+difference comes from the way you configure Pods and other objects to
+consume those values.
+
```

Comment:
It would be good to hint at the next step here for people (Maybe a subheading, `## Using ConfigMaps`?) – just a quick overview along the lines of "Once you've defined a ConfigMap, you can reference the keys in `data` in your Pod's configuration, hurray!"

As it is, I feel like I understand ConfigMaps and what their purpose is, but I'm not sure exactly how to use them :)

Reply:
Thinking about this in more detail: I think it's because in the context of this topic, "data" could either be the data a container is processing _or_ the `data` element in a ConfigMap, and I'm not sure which you're talking about in this sentence.